backend/src/ml_space_lambda/app_configuration/lambda_functions.py

```diff
@@ -18,39 +18,234 @@
+        ],
+        "ResourceType": ResourceType.EMR_CLUSTER,
+    },
+}
 
-    new_configuration = SettingsModel.from_dict(request["configuration"])
+# Instead of deleting the policy (which requires unattaching from all roles) assign a filler non-impactful statement
+filler_deny_statement = (
+    {
+        IAMStatementProperty.EFFECT: IAMEffect.DENY,
+        IAMStatementProperty.ACTION: ["dynamodb:CreateTable"],
+        IAMStatementProperty.RESOURCE: get_account_arn("dynamodb", "table/invalid-table"),
+    },
+)
 
-    app_configuration = AppConfigurationModel(
-        configScope=configScope,
-        version_id=version_id,
-        configuration=new_configuration,
-        changed_by=event["requestContext"]["authorizer"]["principalId"],
-        change_reason=request["changeReason"],
-        created_at=time.time(),
-    )
 
+@event_wrapper
+def update_configuration(event, context):
     try:
-        # Create a record for this configuration and mark it the latest config by incrementing the version Id
-        app_configuration_dao.create(config=app_configuration)
-    except Exception as e:
-        if e.response["Error"]["Code"] == "ConditionalCheckFailedException":
-            e.response["ResponseMetadata"]["HTTPStatusCode"] = 429
-        raise e
+        response_status_code = 200
+        warning_issues = []
+        request = json.loads(event["body"])
+        configScope = request["configScope"]
+        version_id = request["versionId"] + 1  # increment so this will be the latest version
+
+        new_configuration = SettingsModel.from_dict(request["configuration"])
+
+        app_configuration = AppConfigurationModel(
+            configScope=configScope,
+            version_id=version_id,
+            configuration=new_configuration,
+            changed_by=event["requestContext"]["authorizer"]["principalId"],
+            change_reason=request["changeReason"],
+            created_at=time.time(),
+        )
+
+        try:
+            # Create a record for this configuration and mark it the latest config by incrementing the version Id
+            app_configuration_dao.create(config=app_configuration)
+        except Exception as e:
+            if e.response["Error"]["Code"] == "ConditionalCheckFailedException":
+                e.response["ResponseMetadata"]["HTTPStatusCode"] = 429
+            raise e
 
-    return f"Successfully updated configuration for {configScope}, version {version_id}."
+        # Create the new app deny policy
+        try:
+            deny_policy_statements = []
+            enabled_services_dict = app_configuration.configuration.enabled_services.to_dict()
+            # Check each service
+            for service in enabled_services_dict.keys():
+                # If the service has permissions that need to be denied
+                if service in service_disable_permissions:
+                    # If the service is deactivated
+                    if not enabled_services_dict[service]:
```

Comment:
wont this always be false since we are doing `for service in enabled_services_dict.keys():`?

Reply:
Here is the structure:
```
enabled_services_dict = {
  batchTranslate: True,
  realtimeTranslate: False,
  etc...
}

service_disable_permissions = {
    ServiceType.BATCH_TRANSLATE: {
        "Statements": [
            {
                IAMStatementProperty.EFFECT: IAMEffect.DENY,
            ...
       ]
       "ResourceType": ResourceType.BATCH_TRANSLATE_JOB
  },
  ...
}
```

First we check if there are instructions in `service_disable_permissions` for how to deactivate that service, then we check if the service is deactivated in the newly submitted config `if not enabled_services_dict[service]`. For rexample with 'realtimeTranslate' it would be: `if not enabled_services_dict['realtimeTranslate']` == `if not False` == True

Then if the service is deactivated we will apply the deactivation statements.

I will condense this though into a single conditional and check if the service is deactivated first

Reply:
Updated: `if not enabled_services_dict[service] and service in service_disable_permissions:`